drivers/performance_measurement/spec/models/performance_measurement/pm_spm_alignment_spec.rb

```diff
@@ -0,0 +1,311 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require_relative '../../../../hud_spm_report/spec/models/fy2024/shared_context'
+
+RSpec.describe 'Performance Measurement and SPM Alignment', type: :model do
+  include_context 'SPM test setup'
+
+  # Shared context for running both reports with the same data
+  # Shared context for running both reports with the same data
+  shared_context 'report generation' do
+    before do
+      # Generate the service history records for enrollments
+      GrdaWarehouse::Tasks::ServiceHistory::Enrollment.find_each(&:rebuild_service_history!)
+
+      # Create the report filter
+      spm_report_filter = default_filter.dup
+      spm_report_filter.update(project_ids: project_ids)
+
+      # Step 1: Run the Performance Measurement report
+      @pm_report = PerformanceMeasurement::Report.new(
+        user_id: user.id,
+      )
+      @pm_report.filter = spm_report_filter
+      @pm_report.update_goal_configuration!
+      @pm_report.save!
+      @pm_report.update_goal_configuration!
```

Comment:
This is probably some quirk of the report, but why do we need to call `update_goal_configuation!` twice?

Reply:
removed it, thanks